### PR TITLE
fixed update-all driver (ae)

### PR DESCRIPTION
**Pull request by Paul-St-Young**
corrected handling of species with different mass in update-all driver
added tests for update-all driver on bcc hydrogen
added more observables to bcc test

### CODE REVIEW

File: tests/system/CMakeLists.txt [hunk 1]
```diff
@@ -323,9 +327,21 @@ ENDIF()
                     0 # VMC
                     TRUE)
 
+  QMC_RUN_AND_CHECK(short-bccH_1x1x1_ae-vmc-all_sdj
```
Comment (markdewing) [inline tests/system/CMakeLists.txt]: Could you add a comment here (and to the other added test) about the focus of the tests (all-electron moves).   The test names are necessarily short, but cryptic.   Plus there's a double use of 'all electron' - 'ae' is the all-electron potential (vs a pseudopotential) and 'all' for all-electron moves (vs. particle-by-particle moves)

File: tests/system/CMakeLists.txt [hunk 2]
```diff
@@ -335,6 +351,15 @@ ENDIF()
...
+                    "${CMAKE_SOURCE_DIR}/tests/solids/bccH_1x1x1_ae"
+                    qmc_short_vmc_dmc_all
+                    qmc_short_vmc_dmc_all.in.xml
+                    1 16
+                    BCC_H_DMC_SCALARS
+                    1 # DMC
+                    TRUE)
+
   IF(NOT QMC_CUDA)
   LIST(APPEND BCC_H_RMC_SCALARS "totenergy" "-1.8420 0.0011")
```
Comment (ye-luo) [inline tests/system/CMakeLists.txt]: Move the two QMC_RUN_AND_CHECK after line 363     IF(NOT QMC_CUDA) .
GPU has no all-electron move support.

File: tests/system/CMakeLists.txt [hunk 3]
```diff
@@ -336,7 +343,28 @@ ENDIF()
...
+# Check VMCUpdateAllWithDrift driver using all-particle updates
+  QMC_RUN_AND_CHECK(short-bccH_1x1x1_ae-vmc-all_sdj
+                    "${CMAKE_SOURCE_DIR}/tests/solids/bccH_1x1x1_ae"
+                    qmc_short_vmc_all
+                    qmc_short_vmc_all.in.xml
+                    1 16
+                    BCC_H_SCALARS
+                    0 # VMC
+                    TRUE)
   LIST(APPEND BCC_H_RMC_SCALARS "totenergy" "-1.8420 0.0011")
```
Comment (markdewing) [inline tests/system/CMakeLists.txt]: It looks like the DMC test got stuck between the RMC energy definition and the RMC test.

**Comment by ye-luo** [comment]
To build up the matrix, we need a list of specification for the tests we have.
We can then filter out the unnecessary redundant tests and add the missing ones.

**Comment by prckent** [comment]
@Paul-St-Young Any thoughts on what is likely working or near working in terms of update-all methods?

**Comment by Paul-St-Young** [comment]
I just added a new test for the update-all VMC driver without drift. My initial pull request included a test for the update-all DMC driver. Currently the update-all drivers work in the case of no pseudopotential. I am not sure about cases with pseudopotential.

Driver selection is currently not very robust. In QMCApp/QMCDriverFactory.cpp, a bit is set to 1 if move="pbyp"
`WhatToDo[UPDATE_MODE]= (update_mode == "pbyp");`
otherwise the bit is set to 0 and the update-all drivers are used. Someone could inadvertently invoke an update-all driver if "pbyp" was misspelled for example.

**Comment by prckent** [comment]
I suspect we'll need to error trap the GPU version, but lets see the overnight tests first.

Merging.

**Pull request merged**